Fw/Cmd/CmdString.hpp

```diff
@@ -1,37 +1,59 @@
-#ifndef FW_CMD_STRING_TYPE_HPP
-#define FW_CMD_STRING_TYPE_HPP
+// ======================================================================
+// @file   CmdString.hpp
+// @author F Prime
+// @brief  A string sized for a command argument
+// ======================================================================
+
+#ifndef FW_CMD_STRING_HPP
+#define FW_CMD_STRING_HPP
 
 #include <FpConfig.hpp>
-#include <Fw/Types/StringType.hpp>
-#include <Fw/Cfg/SerIds.hpp>
+
+#include "Fw/Cfg/SerIds.hpp"
+#include "Fw/Types/StringBase.hpp"
 
 namespace Fw {
 
-    class CmdStringArg : public Fw::StringBase {
-        public:
+class CmdStringArg final : public StringBase {
+  public:
+    enum {
+        SERIALIZED_TYPE_ID = FW_TYPEID_CMD_STR,
+        STRING_SIZE = FW_CMD_STRING_MAX_SIZE,
+        SERIALIZED_SIZE = STRING_SIZE + sizeof(FwSizeStoreType)
+    };
 
-            enum {
-                SERIALIZED_TYPE_ID = FW_TYPEID_CMD_STR,
-                SERIALIZED_SIZE = FW_CMD_STRING_MAX_SIZE + sizeof(FwBuffSizeType)
-            };
+    CmdStringArg() : StringBase() { *this = ""; }
 
-            CmdStringArg(const char* src);
-            CmdStringArg(const StringBase& src);
-            CmdStringArg(const CmdStringArg& src);
-            CmdStringArg();
-            CmdStringArg& operator=(const CmdStringArg& other);
-            CmdStringArg& operator=(const StringBase& other);
-            CmdStringArg& operator=(const char* other);
-            ~CmdStringArg();
+    CmdStringArg(const CmdStringArg& src) : StringBase() { *this = src; }
 
-            const char* toChar() const;
-            NATIVE_UINT_TYPE getCapacity() const ; //!< return buffer size
+    CmdStringArg(const StringBase& src) : StringBase() { *this = src; }
 
-        private:
+    CmdStringArg(const char* src) : StringBase() { *this = src; }
 
-            char m_buf[FW_CMD_STRING_MAX_SIZE];
-    };
+    ~CmdStringArg() {}
+
+    CmdStringArg& operator=(const CmdStringArg& src) {
+        (void)StringBase::operator=(src);
+        return *this;
+    }
+
+    CmdStringArg& operator=(const StringBase& src) {
+        (void)StringBase::operator=(src);
+        return *this;
+    }
+
+    CmdStringArg& operator=(const char* src) {
+        (void)StringBase::operator=(src);
+        return *this;
+    }
+
+    const char* toChar() const { return this->m_buf; }
+
+    StringBase::SizeType getCapacity() const { return sizeof this->m_buf; }
 
-}
+  private:
+    char m_buf[CmdStringArg::STRING_SIZE];
```

Comment:
While we are at it we should change `char` -> `CHAR`.

Reply:
What is the reason for using CHAR instead of char?

Reply:
In Fw/BasicTypes.h, I see this:
```
typedef char CHAR;
```
This isn't a configuration file, so CHAR is not a configurable type. Is it just for style, to make CHAR look like U32, etc.?